src/SFA.DAS.EmployerIncentives.Functions.LegalEntities/Services/LegalEntities/ILegalEntitiesService.cs

```diff
@@ -8,6 +8,7 @@ public interface ILegalEntitiesService
         Task Refresh();
         Task Refresh(int pageNumber, int pageSize);
         Task Add(AddRequest request);
-        Task Remove(RemoveRequest request);            
+        Task Remove(RemoveRequest request);
+        Task SignAgreement(SignAgreementRequest request);
```

Comment:
Can this be moved to an AgreementService whose responsibility is everything to do with Agreements rather than being on the LegalEntitiesService.  Keeps the different concerns separate.